### ISSUE TITLE
@babel/standalone 7.11.6 crash react native iOS app on launch

### ISSUE THREAD

**Issue by bozou** [babel/babel#12124]
## Bug Report

- [ ] I would like to work on a fix!

**Current behavior**
react-native init myproject
cd myproject
npm install --save @babel/standalone
npx react-native run-ios # maybe this step is optional
app should run fine in simulator
add one line of code in App.js:
`
    const Babel = require('@babel/standalone');
`
before 
`
    const App: () => React$Node = () => {
`

open Xcode -> Menu -> Project -> Scheme -> Edit Scheme... -> change Build Configuration to 'Release'
connect iPhone to your MacBook
Run the app 'myproject' on your iPhone (must be on the iPhone, it does NOT crash in simulator).
myproject crashed on launch with the following message in Xcode console:

2020-09-29 20:29:13.536339-0400 myproject[1174:270849] [native] Running application myproject ({
    initialProps =     {
    };
    rootTag = 1;
})
2020-09-29 20:29:13.815830-0400 myproject[1174:271076] [javascript] TypeError: undefined is not an object (evaluating 'r.type=e')
2020-09-29 20:29:13.818053-0400 myproject[1174:271070] [native] Unhandled JS Exception: TypeError: undefined is not an object (evaluating 'r.type=e')
2020-09-29 20:29:13.819941-0400 myproject[1174:271070] *** Terminating app due to uncaught exception 'RCTFatalException: Unhandled JS Exception: TypeError: undefined is not an object (evaluating 'r.type=e')', reason: 'Unhandled JS Exception: TypeError: undefined is not an object (evaluating 'r.type=e'), stack:
qi@417:71439
l@417:73316
<unknown>@417:1611307
v@2:1473
<unknown>@408:149
v@2:1473
<unknown>@6:57
v@2:1473
d@2:875
global code@420:3
'
*** First throw call stack:
(0x1a20d9654 0x1a1dfbbcc 0x104b45ab0 0x104bc2ba4 0x104bc35dc 0x1a20df8c0 0x1a1fb0a70 0x1a1fb1648 0x104b75bc4 0x104b77d58 0x104b77abc 0x1050d6338 0x1050d7730 0x1050de740 0x1050df2e0 0x1050ea6c4 0x1a1df0b74 0x1a1df3740)
libc++abi.dylib: terminating with uncaught exception of type NSException
*** Terminating app due to uncaught exception 'RCTFatalException: Unhandled JS Exception: TypeError: undefined is not an object (evaluating 'r.type=e')', reason: 'Unhandled JS Exception: TypeError: undefined is not an object (evaluating 'r.type=e'), stack:
qi@417:71439
l@417:73316
<unknown>@417:1611307
v@2:1473
<unknown>@408:149
v@2:1473
<unknown>@6:57
v@2:1473
d@2:875
global code@420:3
'
terminating with uncaught exception of type NSException
*** Terminating app due to uncaught exception 'RCTFatalException: Unhandled JS Exception: TypeError: undefined is not an object (evaluating 'r.type=e')', reason: 'Unhandled JS Exception: TypeError: undefined is not an object (evaluating 'r.type=e'), stack:
qi@417:71439
l@417:73316
<unknown>@417:1611307
v@2:1473
<unknown>@408:149
v@2:1473
<unknown>@6:57
v@2:1473
d@2:875
global code@420:3
'
terminating with uncaught exception of type NSException

- [REPL](babeljs.io/repl), [Codesandbox](https://codesandbox.io/s/babel-repl-custom-plugin-7s08o?file=/src/index.js), or GitHub Repo helps!

**Input Code**

```js
none
```

**Expected behavior**
App should run fine.

**Babel Configuration (babel.config.js, .babelrc, package.json#babel, cli command, .eslintrc)**
- Filename: `package.json`

```js
 cat package.json 
{
  "name": "junk",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "start": "react-native start",
    "test": "jest",
    "lint": "eslint ."
  },
  "dependencies": {
    "@babel/standalone": "^7.11.6",
    "react": "16.13.1",
    "react-native": "0.63.2"
  },
  "devDependencies": {
    "@babel/core": "^7.11.5",
    "@babel/runtime": "^7.11.2",
    "@react-native-community/eslint-config": "^2.0.0",
    "babel-jest": "^26.3.0",
    "eslint": "^7.8.1",
    "jest": "^26.4.2",
    "metro-react-native-babel-preset": "^0.63.0",
    "react-test-renderer": "16.13.1"
  },
  "jest": {
    "preset": "react-native"
  }
}
```

**Environment**

```

```
- Babel version(s): v7.11.5
- Node/npm version: node v10.15.3 / nom 6.4.1
- OS: OSX 10.5.6
- Monorepo: no
- How you are using Babel: react-native

**Possible Solution**

**Additional context**
Add any other context about the problem here. Or a screenshot if applicable

**Comment by JLHwung** [comment]
Out of curiosity, what is your use case of running `@babel/standalone` in iOS app? If you want to transpile React JSX, they should be handled by `react-native-cli`.

**Comment by bozou** [comment]
My iOS react-native-based app runs javascript created by user.  User can use await/async.  It used to work fine with react-native 0.57.2.  When I tried to upgrade to RN 0.63, I ran into https://stackoverflow.com/questions/63787013/script-by-eval-became-stuck-on-react-native-version-0-63-1-ios.  I am guessing that there is something wrong between RN and the JavascriptCOre's event loop when async/await are involved.  If I can transpile async/await into generators, the problem goes away.  But since user writes the JS script, I can't use react-native-cli.  I have to run @babel/standalone in my iOS app.

BTW, I found a work around now.  I cloned babel and removed most from packages/babel-standalone/scripts/pluginConfig.json except for the following:

[
  "syntax-async-generators",
  "proposal-async-generator-functions",
  "transform-async-to-generator"
]

and modified packages/babel-standalone/src/index.js accordingly.

Then built @babel/standalone.  The result babel.js does the async/await to generators transformation and does not crash the app.
